packages/stream_video_noise_cancellation/lib/noise_cancellation_audio_processor.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:stream_video/stream_video.dart';
+import 'package:stream_video_noise_cancellation/stream_video_noise_cancellation.dart';
+
+class NoiseCancellationAudioProcessor extends AudioProcessor {
+  final _logger = taggedLogger(tag: 'SV:NoiseCancellationAudioProcessor');
+
+  NoiseCancellationAudioProcessor() {
+    try {
+      StreamVideoNoiseCancellation().registerProcessor();
+    } on UnimplementedError catch (e) {
+      _logger.w(
+        () => 'Noise cancellation is not supported on this platform: $e',
+      );
+    }
+  }
```

Comment:
You're not catching async errors here, because you don't await. I think it's better to put it in a different async method:

```suggestion
  NoiseCancellationAudioProcessor() {
    _init();
  }
  
  Future<void> _init() async {
    try {
      await StreamVideoNoiseCancellation().registerProcessor();
    } on UnimplementedError catch (e) {
      _logger.w(
        () => 'Noise cancellation is not supported on this platform: $e',
      );
    } catch (err, stackTrace) {
      _logger.e(
        () => err.toString(),
      );
    }
  }
```

I also wonder if we shouldn't make an async factory so the caller knows if/when the noise cancelation is initialized.